static/src/js/components/TemplateForm/TemplateForm.jsx

```diff
@@ -147,11 +146,11 @@ const TemplateForm = () => {
       setLoading(false)
     }
 
-    if (id !== 'new' && !draft) {
+    if (id !== 'new' && isEmpty(draft)) {
       setLoading(true)
       fetchTemplate()
     }
-  }, [id])
+  }, []) // Should only fetch a template on page load
```

Comment:
"Page load" here would be the first time the component is rendered, right? If the user goes back to the list, and clicks another template, wouldn't that just be a change in `id`?

Reply:
It should happen every time the component is mounted written like this, which I think is correct. The component is unmounted and remounted any time you navigate away and back.